### PR TITLE
feat: add status to SDK Trials [MLG-668]

### PR DESCRIPTION
**Pull request by azhou-determined**
## Description

adds a status parameter to SDK trial objects. 

## Test Plan

call `get_trial` for an existing trial. verify that the `Trial` returned contains a field `status` with the trial's state.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: harness/determined/common/experimental/trial.py [hunk 1]
```diff
@@ -157,6 +157,7 @@ def __init__(self, trial_id: int, session: api.Session):
 
         self.hparams: Optional[Dict[str, Any]] = None
         self.summary_metrics: Optional[Dict[str, Any]] = None
+        self.status: Optional[bindings.experimentv1State] = None
```
Comment (azhou-determined) [inline harness/determined/common/experimental/trial.py:160]: might be worth making a separate public-facing status object. would basically be duplicated code, but i don't like that we're surfacing bindings to users.

Comment (wes-turner) [reply harness/determined/common/experimental/trial.py:160]: I'm interested here in what @rb-determined-ai thinks about a

```python
from determined.common.api.bindings import experimentv1State
```

and then using that.

File: harness/tests/determined/common/experimental/test_determined.py [hunk 2]
```diff
@@ -126,7 +126,7 @@ def test_default_retry_doesnt_retry_allowed_status(
 
 
 @responses.activate
-def test_get_trial_populates_summary_metrics(make_client: Callable[[], Determined]) -> None:
+def test_get_trial_populates_all_parameters(make_client: Callable[[], Determined]) -> None:
```
Comment (wes-turner) [inline harness/tests/determined/common/experimental/test_determined.py:129]: If this is parameterizable with `pytest.mark.parameterize`, then could you please use that? If not, then [the docs](https://docs.pytest.org/en/7.3.x/how-to/parametrize.html#basic-pytest-generate-tests-example) talk about how to generate more complex parameterizations.

I don't expect it to be difficult, one way or the other. If it is, then let's chat and maybe we can figure something else out.

File: harness/determined/common/experimental/trial.py [hunk 3]
```diff
@@ -388,6 +389,7 @@ def __repr__(self) -> str:
     def _hydrate(self, trial: bindings.trialv1Trial) -> None:
         self.hparams = trial.hparams
         self.summary_metrics = trial.summaryMetrics
+        self.status = trial.state
```
Comment (wes-turner) [inline harness/determined/common/experimental/trial.py:392]: I love how easy it is to add new attributes.

File: harness/determined/common/experimental/trial.py (hunk 1 above)
Comment (rb-determined-ai) [reply harness/determined/common/experimental/trial.py:160]: I am of the opinions that swagger-generated bindings generally suck and are unergonomic, making them undesirable to ever give to end users.  They're basically only for type safety, and I think we should be careful not to leak them to end users.

Specifically in the enum case, proto always generates an UNSPECIFIED enum value which makes sense in the protobuf serialization layer but semantically does not make sense to expose in python.  If UNSPECIFIED were necessary, we should be accepting `Optional[thing]` where `None` serves the purpose of UNSPECIFIED.

**Pull request closed without merging**